src/main/java/io/phasetwo/service/resource/UserResource.java

```diff
@@ -59,6 +66,79 @@ public Stream<OrganizationRole> listUserOrgRoles(
     }
   }
 
+  @PUT
+  @Path("/{userId}/orgs/{orgId}/roles")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response grantUserOrgRoles(@PathParam("userId") String userId, @PathParam("orgId") String orgId,
+                                    List<OrganizationRole> rolesRep) {
+    log.debugv("Grant user organization roles for %s %s %s", realm.getName(), userId, orgId);
+    UserModel user = session.users().getUserById(realm, userId);
+    OrganizationModel org = orgs.getOrganizationById(realm, orgId);
+    canManage(userId, orgId, user, org);
+    rolesRep.forEach(r -> {
```

Comment:
Need a test that validates when some of the grants fail:
- everything fails
- no events get emitted
- a proper error gets sent back